scripts/databricks_install.py

```diff
@@ -0,0 +1,255 @@
+    """
+
+CLUSTER_NOT_RUNNING_MSG = """
+    Cluster {0} found, but it is not running. Status={1}
+    You can start the cluster with 'databricks clusters start --cluster-id {0}'.
+    Then, check the cluster status by using 'databricks clusters list' and
+    re-try installation once the status becomes 'RUNNING'.
+    """
+""
+
+## Variables for operationalization:
+COSMOSDB_JAR_FILE_OPTIONS = {
+  '3': 'https://search.maven.org/remotecontent?filepath=com/microsoft/azure/azure-cosmosdb-spark_2.2.0_2.11/1.1.1/azure-cosmosdb-spark_2.2.0_2.11-1.1.1-uber.jar',
+  '4': 'https://search.maven.org/remotecontent?filepath=com/microsoft/azure/azure-cosmosdb-spark_2.3.0_2.11/1.2.2/azure-cosmosdb-spark_2.3.0_2.11-1.2.2-uber.jar',
+  '5': 'https://search.maven.org/remotecontent?filepath=com/microsoft/azure/azure-cosmosdb-spark_2.4.0_2.11/1.3.5/azure-cosmosdb-spark_2.4.0_2.11-1.3.5-uber.jar'
+}
+
+PYPI_O16N_LIBS=["azure-cli==2.0.56", "azureml-sdk[databricks]==1.0.8", "pydocumentdb==2.3.3"]
+
+
+def create_egg(path_to_recommenders_repo_root=os.getcwd(),local_eggname='Recommenders.egg', overwrite=False):
+  """
+  Packages files in the reco_utils directory as a .egg file that can be uploaded to dbfs and installed as a library on a databricks cluster.
+  """
+  ## create the zip archive:
+  myzipfile = shutil.make_archive('reco_utils',
+                'zip', 
+                root_dir = path_to_recommenders_repo_root, 
+                base_dir = 'reco_utils'
+              )
+
+  ## make sure extension is .egg
+  if local_eggname[-4:] != ".egg":
+    print('appending .egg to end of name.')
+    logal_eggname = local_eggname+'.egg'
+  
+  ## overwrite egg if it previously existed
+  if os.path.exists(local_eggname) and overwrite:
+    os.unlink(local_eggname)
+  os.rename(myzipfile,local_eggname)
+  return local_eggname
+
+def prepare_for_operationalization(cluster_id, api_client, status=None, dbfs_path="dbfs:/FileStore/jars", overwrite=False):
+  """
+  Installs appropriate versions of several libraries to support operationalization.
+  """
+  print("Preparing for operationliazation...")
+  if status is None:
+    ## get status if None
+    status=ClusterApi(my_api_client).get_cluster(args.cluster_id)
+  
+  spark_version=status['spark_version'][0]
+  cosmosdb_jar_url = COSMOSDB_JAR_FILE_OPTIONS[spark_version]
+
+  ## download the cosmosdb jar
+  local_jarname = os.path.basename(cosmosdb_jar_url)
+  ## only download if you need it:
+  if overwrite or not os.path.exists(local_jarname):
+    print('Downloading {}...'.format(cosmosdb_jar_url))
+    local_jarname, _ = urlretrieve(cosmosdb_jar_url,local_jarname)
+  else:
+    print('File {} already downloaded.'.format(local_jarname))
+
+  ## upload to dbfs:
+  uploadpath='/'.join([dbfs_path, local_jarname])
+  print("Uploading CosmosDB driver to databricks at {}".format(uploadpath))
+  DbfsApi(api_client).cp(recursive=False, src=local_jarname, dst=uploadpath, overwrite=overwrite)
```

Comment:
is there any way to know if the file existed and was not overwritten after this call? might be useful to inform the user if they reran it without the overwrite flag

Reply:
it fails if the file exists and overwrite=False. Added a helper function to alert the user to when it is being overwritten., we could also change behavior if we want to allow it to continue if the file exists.